src/librustc_trans/collector.rs

```diff
@@ -433,6 +434,40 @@ fn check_recursion_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     (instance.def, recursion_depth)
 }
 
+fn check_type_length_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
I actually expected this value to be stored inside each `TyS` - seems cheaper that way?